monitoring/node_collector.go

```diff
@@ -49,33 +59,90 @@ func NewNodeCollector(
 		Type: prometheus.GaugeValue,
 	}
 
+	nc.cpuUsageMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemNode, "cpu_usage_millicpu"),
+			"The cpu usage on this node",
+			[]string{nodeLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	nc.cpuCapacityMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemNode, "cpu_capacity_millicpu"),
+			"The maximum allocatable cpu on this node",
+			[]string{nodeLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	nc.memoryUsageMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemNode, "memory_usage_bytes"),
+			"The memory usage on this node",
+			[]string{nodeLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	nc.memoryCapacityMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemNode, "memory_capacity_bytes"),
+			"The maximum allocatable memory on this node",
+			[]string{nodeLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
 	return nc
 }
 
 func (nc *NodeCollector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- nc.statusMetric.Desc
 	ch <- nc.totalNumberOfNodesMetric.Desc
+	ch <- nc.cpuUsageMetric.Desc
+	ch <- nc.cpuCapacityMetric.Desc
+	ch <- nc.memoryUsageMetric.Desc
+	ch <- nc.memoryCapacityMetric.Desc
 }
 
 func (nc *NodeCollector) Collect(ch chan<- prometheus.Metric) {
-	nodeList, err := nc.ds.ListNodesRO()
-	if err != nil {
-		nc.logger.WithError(err).Warn("error during scrape")
-		return
-	}
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go nc.collectNodeStatus(ch, &wg)
+
+	wg.Add(1)
+	nc.collectTotalNumberOfNodes(ch, &wg)
```

Comment:
hmm? Those are single-threaded so why you need waitgroup? Did you mean to call those as goroutines?

Also, you can avoid passing the waitgroup into the function, since waitgroup is just a flow control e.g.
``` 
wg.Add(1)
go func() {
    defer wg.Done()
    call_func()
}()
```

Reply:
My bad. Yes I meant to call those funcs as goroutines. Also fixed to avoid passing the waitgroup.